src/bin/pg_autoctl/string_utils.c

```diff
@@ -0,0 +1,434 @@
+/*
+ * src/bin/pg_autoctl/string_utils.c
+ *   Implementations of utility functions for string handling
+ *
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the PostgreSQL License.
+ *
+ */
+
+#include <errno.h>
+#include <inttypes.h>
+#include <limits.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+
+#include "file_utils.h"
+#include "string_utils.h"
+
+/*
+ * intToString converts an int to an IntString, which contains a decimal string
+ * representation of the integer.
+ */
+IntString
+intToString(int64_t number)
+{
+	IntString intString;
+
+	intString.intValue = number;
+
+	sformat(intString.strValue, INTSTRING_MAX_DIGITS, "%" PRId64, number);
+
+	return intString;
+}
+
+
+/*
+ * converts given string to 64 bit integer value.
+ * returns 0 upon failure and sets error flag
+ */
+bool
+stringToInt(const char *str, int *number)
+{
+	char *endptr;
+	bool success = true;
+	long long int n;
+
+	if (number == NULL)
+	{
+		return false;
+	}
+
+	errno = 0;
+
+	n = strtoll(str, &endptr, 10);
+
+	if (str == endptr)
+	{
+		success = false;
```

Comment:
This success variable is not really needed. It makes the code easier if we simply return early.
```suggestion
		return false;
```

Reply:
changed